pkg/multiclustermanager/routes.go

```diff
@@ -111,9 +110,21 @@ func router(ctx context.Context, localClusterEnabled bool, tunnelAuthorizer *mcm
 	authed.PathPrefix("/v3/token").Handler(tokenAPI)
 	authed.PathPrefix("/v3").Handler(managementAPI)
 
+	// Authenticated routes
+	metricsAuthed := mux.NewRouter()
+	metricsAuthed.UseEncodedPath()
+	impersonsatingAuth := auth.ToMiddleware(requests.NewImpersonatingAuth(sar.NewSubjectAccessReview(clusterManager)))
+	tokenReviewAuth := impersonsatingAuth.Chain(auth.ToMiddleware(requests.NewTokenReviewAuth(scaledContext.K8sClient.AuthenticationV1())))
+	metricsAuthed.Use(mux.MiddlewareFunc(tokenReviewAuth))
+	metricsAuthed.Use(mux.MiddlewareFunc(rbac.NewAccessControlHandler()))
+	metricsAuthed.Use(requests.NewAuthenticatedFilter)
+
+	metricsAuthed.Path("/metrics").Handler(metricsHandler)
+
 	unauthed.NotFoundHandler = authed
 	return func(next http.Handler) http.Handler {
-		authed.NotFoundHandler = next
+		authed.NotFoundHandler = metricsAuthed
```

Comment:
Why is this needed?

Reply:
This chains the routing logic together, first it looks for routes from the unauth handler, if nothing matches then from auth, if nothing matches then from metrics auth. This is also relates to your two above questions. If metrics matches in the auth handler it will not be forwarded to the metrics handler. Using a new auth handler is how we are only using this auth for metrics otherwise we could use less code and just use the auth handler with metrics endpoint but then that performs the metrics auth logic on everything that fails first auth check.

Reply:
Since this line does not use the `next` var, can we move it outside of this function (just like how we set `unauthed.NotFoundHandler`? I think it only needs to be set once.